### ISSUE TITLE
Vulnerability false positives

### ISSUE THREAD

**Issue by MattShip** [wazuh/wazuh#27235]
|Wazuh version|Component|Install type|Install method|Platform|
| 4.9.2 | Wazuh Vulnerability Detection | Manager | Local Install | Debian 6.1.119-1 |

Hi, I've recently installed Wazuh and when looking at the vulnerabilities I've noticed what I believe to be some false positives, as follows. I hope the screenshots below show sufficient information given that they show the package version on my server as well as CVE id. If you need any further info let me know.

CVE-2024-26952, fixed on Debian 6.1.119-1 according to https://security-tracker.debian.org/tracker/CVE-2024-26952 
![Screenshot 2024-12-09 at 14 20 49](https://github.com/user-attachments/assets/4e4c2501-6bc9-485c-8dc7-a178ea81592a)

CVE-2022-1271, fixed in gzip 1.12-1 according to https://security-tracker.debian.org/tracker/CVE-2022-1271
![Screenshot 2024-12-09 at 14 21 35 1](https://github.com/user-attachments/assets/e448331d-d8a5-4138-b82d-40bdc0ee7926)

CVE-2023-2603, fixed in libcap2 1:2.66-4 according to https://security-tracker.debian.org/tracker/CVE-2023-2603
![Screenshot 2024-12-09 at 14 34 05](https://github.com/user-attachments/assets/9588eda9-8269-4bf1-b24c-71b897daa03f)

CVE-2024-6716, no reference in debian security tracker, looks like this has been rejected according to https://nvd.nist.gov/vuln/detail/CVE-2024-6716
![Screenshot 2024-12-09 at 14 35 21](https://github.com/user-attachments/assets/796ade51-05eb-40de-914d-7b2f5c10a5ba)

**Comment by jftuduri** [comment]
Hi @MattShip,

Upon initial review, these do appear to be false positives. I will investigate our vulnerability content generation process to identify the root cause and update you once I have more information.

In the meantime, could you please share the full OS information for the agent? You can retrieve this from `Server Management -> Dev Tools` using the following query (replacing <AGENT_ID> with the appropriate value):

```
GET /syscollector/<AGENT_ID>/os
```

Regards!

**Comment by MattShip** [comment]
Hi @jftuduri

Thanks for getting back to me so promptly.

I've attached the response to that query below

{
  "data": {
    "affected_items": [
      {
        "os": {
          "codename": "bookworm",
          "major": "12",
          "name": "Debian GNU/Linux",
          "platform": "debian",
          "version": "12 (bookworm)"
        },
        "scan": {
          "id": 0,
          "time": "2024-12-10T09:16:42+00:00"
        },
        "release": "6.1.0-28-cloud-amd64",
        "sysname": "Linux",
        "version": "#1 SMP PREEMPT_DYNAMIC Debian 6.1.119-1 (2024-11-22)",
        "architecture": "x86_64",
        "hostname": "*********",
        "agent_id": "007"
      }
    ],
    "total_affected_items": 1,
    "total_failed_items": 0,
    "failed_items": []
  },
  "message": "All specified syscollector information was returned",
  "error": 0
}

**Comment by jftuduri** [comment]
Hi @MattShip,
After further investigation, we’ve confirmed that these are false positives. We are working on addressing them, though the time required for resolution may vary depending on the root cause and the internal processes involved. Once the content is updated, your manager will automatically update its internal vulnerability database, trigger a re-scan, and these vulnerabilities should no longer be reported.

- `CVE-202-26952`: Our content appears outdated, reflecting the state before the fix was released. We will update it soon.
- `CVE-2022-1271`, `CVE-2023-2603`: Our content lists the fixed versions as 1.12-1+b2 and 1:2.66-4+b2, respectively. This discrepancy arises from a process that converts source packages into binary packages for vulnerability detection. We are actively working to resolve this issue.
- `CVE-2024-6716`: It seems Debian initially included this CVE in their feed but later removed it. Our content has not yet been updated to reflect this change. We will address this shortly.

Thanks for reporting this issues.
Regards!

**Comment by linaia** [comment]
Hi, I put here another possible false positive alert on Debian 11 (bookworm) servers (wazuh 4.10.0-1)

https://security-tracker.debian.org/tracker/CVE-2020-14343

python3-yaml is already the newest version (5.3.1-5).

**Comment by jftuduri** [comment]
Hello @linaia!

Could you please provide a screenshot of the Vulnerabilities inventory where the CVE ID, package.name and package.version are visible?
Also, please share the output of the following commands, you should go to Server management -> Dev Tools:
(Replace <AGENT_ID> for  the correct ID)

`GET /syscollector/<AGENT_ID>/os`

`GET /syscollector/<AGENT_ID>/packages?name=python3-yaml`

I just checked our vulnerability content and I see that it is correct: for Debian `bookworm` the package `python3-yaml` is vulnerable if the version it `earlier than 5.3.1-4`. So we need this extra information to analyze this case.

Regards

**Comment by jftuduri** [comment]
Hi @linaia,

It seems that there are two versions of PyYAML installed on the system.

From the screenshot, we can see that the vulnerability is reported for the package with:
- Name: PyYAML
- Version: 5.3.1 (vulnerable)
- Type: pypi (most likely managed by a Python package manager like pip).

However, the query to the packages API was filtered for the package name `python3-yaml`, which is why we only saw the system-installed version of the package.

If you run this query instead:
```
GET /syscollector/<AGENT_ID>/packages?search=yaml
```
You should be able to see both packages listed. Alternatively, you can also verify this in the agent's Inventory Data under Packages.
<details><summary>Example Inventory Data</summary>

![Image](https://github.com/user-attachments/assets/3c65a91d-78b6-4692-ac4d-bc3ff340c7c0)

</details>

In summary:

- The package `python3-yaml v5.3.1-5` it is not reporting any vulnerability, which is correct.
- The vulnerability `CVE-2020-14343` is being reported for the package `PyYAML` `v5.3.1`, also correct.

One important thing to mention here is that the vulnerability scanner uses multiple feeds to determine vulnerabilities, for Debian packages it uses data from the [Debian Security Tracker](https://security-tracker.debian.org), but when analyzing python packages (package.type=pypi) it uses PyPI specific data from [OSV.dev](https://osv.dev).
In this case, OSV.dev indicates the the fix for the PyYAML vulnerability is available in version 5.4: https://osv.dev/vulnerability/GHSA-8q59-q68h-6hv4

Let me know if you need further clarification!

**Comment by linaia** [comment]
On the server, there is no other pyYaml installed : 

````
apt list --installed | grep yaml  
libyaml-0-2/oldstable,now 0.2.2-1 amd64 [installed,automatic]
libyaml-dev/oldstable,now 0.2.2-1 amd64 [installed]
python3-yaml/oldstable,now 5.3.1-5 amd64 [installed,automatic]
```

Result of GET /syscollector/<AGENT_ID>/packages?search=yaml

```
{
  "data": {
    "affected_items": [
      {
        "scan": {
          "id": 0,
          "time": "2024-02-07T11:45:46+00:00"
        },
        "vendor": " ",
        "description": " ",
        "source": " ",
        "architecture": " ",
        "name": "PyYAML",
        "format": "pypi",
        "install_time": " ",
        "version": "5.3.1",
        "section": " ",
        "location": "/usr/lib/python3/dist-packages/PyYAML-5.3.1.egg-info",
        "priority": " ",
        "size": 0,
        "agent_id": "001"
      },
      {
        "scan": {
          "id": 0,
          "time": "2025-01-13T13:13:19+00:00"
        },
        "multiarch": "allowed",
        "vendor": "Debian Python Team <team+python@tracker.debian.org>",
        "description": "YAML parser and emitter for Python3",
        "source": "pyyaml",
        "architecture": "amd64",
        "name": "python3-yaml",
        "format": "deb",
        "install_time": " ",
        "version": "5.3.1-5",
        "section": "python",
        "location": " ",
        "priority": "optional",
        "size": 542720,
        "agent_id": "001"
      }, ...
// 2 other packages libyaml-dev and libyaml-0-2
```

As you can see this is an old scan : "time": "2024-02-07T11:45:46+00:00" and the package is no more on the server and a recent one with the correct installed package -> how to remove this glitch from all the agent data (I have many servers with this CVE alert on python3-yaml)

**Comment by jftuduri** [comment]
That date is indeed quite old. 
However, I just tried installing python3-yaml v5.3.1-5 on a fresh bullseye and the pypi module PyYAML v5.3.1 was installed automatically, so it looks like it comes bundled with the Debian package. 
Could you please verify if you can access this file: /usr/lib/python3/dist-packages/PyYAML-5.3.1.egg-info?

**Comment by linaia** [comment]
Hi, 
I have access to the file : 

```
ls -al /usr/lib/python3/dist-packages/PyYAML-5.3.1.egg-info
-rw-r--r-- 1 root root 1781 Jun 25  2021 /usr/lib/python3/dist-packages/PyYAML-5.3.1.egg-info
```

I think this is rather an old scan that was not cleaned, so how to clean it ?

**Comment by jftuduri** [comment]
Hi @linaia,

The presence of the file on your system indicates that it is not from an old scan. I checked with the team and the date we see in the syscollector API is updated whenever there is a change in the package's information. It does not reflect the date of the last scan of the system.

It's important to note that the reported vulnerability is for the Python package `PyYAML` v3.5.1, not `python3-yaml`. That said, it looks like the `python3-yaml` package includes and installs `PyYAML` as a dependency.

If upgrading the version of `PyYAML` included with `python3-yaml` is not feasible, you could consider removing the `python3-yaml` package and installing a newer version of `PyYAML` directly via pip. Of course, this approach depends on your specific use case and any system restrictions.

**Comment by Damian-Mangold** [comment]
Sanitizaton for CVE-2024-6716 applied.

Regarding vulnerability detection in python packages, we will use this [issue](https://github.com/wazuh/wazuh/issues/27704) to follow the resolution of the problem.

Please reopen it if necessary.
